exponentation_by_squaring/exponentation_by_squaring.c

```diff
@@ -0,0 +1,25 @@
+#include<stdio.h>
+
+/* Algorithm to calculate x raised to power k very efficiently
+*  Can be used as a great alternative to normal pow function
+*  base - base of the expression
+*  power - power of expression
+*/
+
+long int exponentation_by_squaring(int base, int power) {
```

Comment:
@singhpratyush The return type `long double` is not giving answer for any range of inputs I tested but making it double works for negative and positive both the cases.